File: KBDTypeFront/src/dto/order/OrderShowDto.ts
export class OrderShowDto {
    userId: number = 0;
    id: number = 0;    
    firstName: string = '';
    lastName: string = '';
    email?: string;
    phoneNumber: string = '';
    region: string = '';
    city: string = '';
    street: string = '';
    building: string = '';
    apartment?: string;
    postalCode: string = '';
    comment?: string;
    items: {
        productId: number;
        quantity: number;
        price: number;
        productName: string | null; // Optional, can be null if product not found
        imageUrl: string; // URL to the product image
    }[] = [];
    paymentId?: number;
    trackingNumber?: string;
    status: number = 0;
    subtotal: number = 0;
    totalPrice: number = 0;
    createdAt: Date = new Date();
    paidAt?: Date;
    shippedAt?: Date;
    deliveredAt?: Date;
    cancelledAt?: Date;
    refundedAt?: Date;
    
    constructor(init?: Partial<OrderShowDto>) {
        Object.assign(this, init);
    }
}

File: KBDTypeServer/Application/DTOs/OrderDtos/OrderShowDto.cs
using KBDTypeServer.Domain.Enums;

namespace KBDTypeServer.Application.DTOs.OrderDtos
{
    public class OrderShowDto
    {
        public int? UserId { get; set; }
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string? Email { get; set; }
        public string PhoneNumber { get; set; }
        public string Region { get; set; }
        public string City { get; set; }
        public string Street { get; set; }
        public string Building { get; set; }
        public string? Apartment { get; set; }
        public string PostalCode { get; set; }
        public string? Comment { get; set; }
        public List<OrderItemShowDto> Items { get; set; }
        public string? PaymentId { get; set; }
        public string? TrackingNumber { get; set; }
        public OrderStatus Status { get; set; }
        public decimal Subtotal { get; set; }
        public decimal TotalPrice { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? PaidAt { get; set; }
        public DateTime? ShippedAt { get; set; }
        public DateTime? DeliveredAt { get; set; }
        public DateTime? CancelledAt { get; set; }
        public DateTime? RefundedAt { get; set; }
   
    }
}
